### ISSUE TITLE
NotImplementedError on Sync

### ISSUE THREAD

**Issue by jbward1** [toluaina/pgsync#188]
PGSync version: 2.1.6

Postgres version: 

Elasticsearch version: 7.14.1

Redis version: 6.2.5

Python version: 3.8

Problem Description: 
I have a simple schema where a parent has one to many children. On starting the daemon, everything syncs successfully, but when I add additional child elements I'm getting a `ERROR:pgsync.sync: Exception` `NotImplementedError`.  

For more details on project setup, please see: https://github.com/jbward1/pgsync-poc

Error Message (if any): 

```
Syncing datasets db: [0] => redis: [0] => elastic: [0]...
2021-10-01 19:12:11.367:ERROR:pgsync.sync: Exception 
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 876, in sync
    self.es.bulk(self.index, docs)
  File "/usr/local/lib/python3.8/site-packages/pgsync/elastichelper.py", line 111, in bulk
    for _ in helpers.parallel_bulk(
  File "/usr/local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 472, in parallel_bulk
    for result in pool.imap(
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 144, in _helper_reraises_exception
    raise ex
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 388, in _guarded_task_generation
    for i, x in enumerate(iterable):
  File "/usr/local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 155, in _chunk_actions
    for action, data in actions:
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 719, in _payloads
    yield from self._sync(filters=filters, extra=extra)
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 786, in _sync
    self._build_filters(filters, node)
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 754, in _build_filters
    [
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 756, in <listcomp>
    getattr(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 416, in python_type
    raise NotImplementedError()
NotImplementedError
Exception in thread Thread-16:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 905, in poll_redis
    self.on_publish(payloads)
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 976, in on_publish
    self.sync(self._payloads(_payload))
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 876, in sync
    self.es.bulk(self.index, docs)
  File "/usr/local/lib/python3.8/site-packages/pgsync/elastichelper.py", line 111, in bulk
    for _ in helpers.parallel_bulk(
  File "/usr/local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 472, in parallel_bulk
    for result in pool.imap(
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 144, in _helper_reraises_exception
    raise ex
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 388, in _guarded_task_generation
    for i, x in enumerate(iterable):
  File "/usr/local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 155, in _chunk_actions
    for action, data in actions:
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 719, in _payloads
    yield from self._sync(filters=filters, extra=extra)
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 786, in _sync
    self._build_filters(filters, node)
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 754, in _build_filters
    [
  File "/usr/local/lib/python3.8/site-packages/pgsync/sync.py", line 756, in <listcomp>
    getattr(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 416, in python_type
    raise NotImplementedError()
NotImplementedError
Syncing datasets db: [2] => redis: [2] => elastic: [2]...
```

**Comment by toluaina** [comment]
For the shared repo, should I just do
`docker-compose up`

?

**Comment by jbward1** [comment]
@toluaina yep, just docker compose up

**Comment by enterpinamullah** [comment]
Thats great to here! I already installed from master, In my issues everything has been resolved Sir, but only that bootstrap takes 30mins to complete!

**Comment by jbward1** [comment]
@toluaina looks good! Anyway we could get a version published soon?

**Comment by toluaina** [comment]
This had been published as 2.1.7

**Comment by toluaina** [comment]
closing this now thanks
